cmake/HPX_SetupVc.cmake

```diff
@@ -72,11 +76,13 @@ foreach(_flag ${Vc_DEFINITIONS})
   set_property(TARGET hpx::vc APPEND PROPERTY INTERFACE_COMPILE_DEFINITIONS ${_flag})
```

Comment:
Should we also do the same for `target_compile_options` and `target_compile_definitions` ?

Reply:
I don't know. I changed the `target_link_libraries` because the old code causes issues on multi-build-type platforms. The others don't cause any issues (for us), so I'm impartial to those changes.